### DISCUSSION TITLE
How to increase output depth map resolution?

### DISCUSSION THREAD

**Discussion by oscarqht** [isl-org/ZoeDepth#22] [Ideas]
Hi. Great work and thanks for sharing the colab notebook! I noticed that the on the colab notebook, output depth map appears to be of relatively low resolution compared to the input image, which could result in some fuzziness. As such, I am curious to know whether there is a way to increase the output depth map resolution.

If it is not too much trouble, could you kindly point me to the relevant section of the notebook where I can make the necessary changes? I would be incredibly grateful for your help in this regard :)

![image](https://user-images.githubusercontent.com/165961/222314040-39031497-207c-4886-9cd4-387ce8b90cba.png)

**Comment by talasalim** [comment]
I also used the notebook, do you know how we can add code to extract the metric depth at each pixel?

**Comment by razorx89** [comment]
I am also curious how we could improve on the depth map resolution. I think the current pre-trained models won't be able to generate higher resolutions. However, are there any plans to use backbones with larger image sizes? Or are there any tricks to generate e.g. a sliding window inference with recalibration based on other predictions? For example,  I cropped a high resolution image into multiple non-overlapping images and stitched the resulting depth maps together, but the boundary artifacts are clearly visible and there are also some depth inconsistencies between the individual crops.

A hierarchical approach could be the way to go, using ZoeDepth for depth estimation at the full image but coarse resolution and afterwards using this as an additional input to give guidance to a decoder which predicts on a fine resolution crop.

**Comment by oscarqht** [comment]
[Boosting Monocular Depth Estimation Models to High-Resolution via Content-Adaptive Multi-Resolution Merging – Yağız Aksoy](http://yaksoy.github.io/highresdepth/) this research looks like a good place to start @razorx89

**Reply by BillFSmith** [reply]
The https://github.com/compphoto/BoostYourOwnDepth work is great and can be used with ZoeDepth, but I found it gave banding, which shows up as pixelation in the depth axis when converting the depth map to a 3d model.

**Comment by BillFSmith** [comment]
I've written a [proof of principle colab script](https://github.com/BillFSmith/TilingZoeDepth) to increase the apparent output resolution from ZoeDepth (which already gives amazing results). An example is shown below

Original:
![zoe_depth_map_16bit_low(2)](https://github.com/isl-org/ZoeDepth/assets/66475393/135a03e3-eb65-480c-99bc-f2bde7deb26e)

High resolution using tiling and combining:
![combined_image(8)](https://github.com/isl-org/ZoeDepth/assets/66475393/b74cbffd-19c1-498e-8a2b-057c0f247483)

**Reply by shariqfarooq123** [reply]
This looks great! Thanks for sharing your work!

**Reply by oscarqht** [reply]
This is amazing!!

**Reply by razorx89** [reply]
Great work, thank you! 👍

**Reply by kwea123** [reply]
the idea is good, but it becomes non-metric after your operations.

**Reply by codikzng** [reply]
Great work! Is this what you came up with, or are there any similar research findings?
